packages/mdxts/src/utils/get-git-file-url.ts

```diff
@@ -1,35 +1,41 @@
+import { join } from 'path'
+
 /** Constructs a URL for a specific file in a Git repository. Supports GitHub, GitLab, and Bitbucket. */
 export function getGitFileUrl(
   filePath: string,
   line: number = 0,
   column: number = 0,
   gitSource: string,
-  gitBranch: string = 'main'
+  gitBranch: string = 'main',
+  gitProvider?: string
 ): string {
   const url = new URL(gitSource)
   let fileUrl: string
 
-  switch (url.hostname) {
-    case 'github.com':
-      fileUrl = `${gitSource}/blob/${gitBranch}/${filePath}`
+  const provider =
+    gitProvider || gitSource.replace(/^https?:\/\/(?:.*\.)?(.*?)\..*/gm, `$1`)
```

Comment:
alternative would be using a package which resolves the domain name - but for our case, I think a "simple" regex should be enough

Reply:
Yeah, simple is great! It can always be adapted later if needed.